target/linux/lantiq/files/arch/mips/boot/dts/lantiq/ar10_tplink_archer-d2.dts

```diff
@@ -0,0 +1,268 @@
+                        linux,default-trigger = "usbport";
+                };
+
+                lan {
+                        label = "lan";
+                        gpios = <&gpio 8 GPIO_ACTIVE_LOW>;
+                };
+
+                led_wlan: wlan {
+                        label = "wlan";
+                        gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+                };
+
+                wps {
+                        label = "wps";
+                        gpios = <&gpio 27 GPIO_ACTIVE_LOW>;
+                };
+        };
+
+        usb_vbus: regulator-usb-vbus {
+		compatible = "regulator-fixed";
+
+		regulator-name = "USB_VBUS";
+
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+
+		gpio = <&gpio 25 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+};
+
+&spi {
+        status = "okay";
+
+        flash@0 {
+                compatible = "jedec,spi-nor";
+                reg = <1>;
+                spi-max-frequency = <25000000>;
+                m25p,fast-read;
+
+                partitions {
+                        compatible = "fixed-partitions";
+                        #address-cells = <1>;
+                        #size-cells = <1>;
+
+                        partition@0 {
+                                reg = <0x0 0x20000>;
+                                label = "u-boot";
+                                read-only;
+                        };
+
+                        partition@20000 {
+                                reg = <0x20000 0xf90000>;
+                                label = "firmware";
+                        };
+
+                        partition@fb0000 {
+                                reg = <0xfb0000 0x10000>;
+                                label = "radioDECT";
+                                read-only;
+                        };
+
+                        partition@fc0000 {
+                                reg = <0xfc0000 0x10000>;
+                                label = "config";
+                                read-only;
+                        };
+
+                        romfile: partition@fd0000 {
+                                reg = <0xfd0000 0x10000>;
+                                label = "romfile";
+                                read-only;
+                        };
+
+                        partition@fe0000 {
+                                reg = <0xfe0000 0x10000>;
+                                label = "rom";
+                                read-only;
+                        };
+
+                        radio: partition@ff0000 {
+                                reg = <0xff0000 0x10000>;
+                                label = "radio";
+                                read-only;
+                        };
+                };
+        };
+};
+
+&eth0 {
+        pinctrl-names = "default";
+	mtd-mac-address = <&romfile 0xf100>;
+};
+
+&gswip {
+	pinctrl-0 = <&mdio_pins>;
+	pinctrl-names = "default";
+};
+
+//&gphy0 {
```

Comment:
You can keey this. This control which firmware should be loaded to PHY:
* GPHY_MODE_GE - Gibabit firmware
* GPHY_MODE_FE - Fast Ethernet firmware